lms/djangoapps/oauth2_handler/handlers.py

```diff
@@ -192,17 +193,21 @@ def find_courses(self, user, access_type, values=None):
 
     # pylint: disable=missing-docstring
     def _get_courses_with_access_type(self, user, access_type):
-
         # Check the application cache and update if not present. The application
         # cache is useful since there are calls to different endpoints in close
-        # succession, for example the id_token and user_info endpoins.
+        # succession, for example the id_token and user_info endpoints.
 
         key = '-'.join([str(self.__class__), str(user.id), access_type])
         course_ids = cache.get(key)
 
-        if course_ids is None:
-            course_ids = [unicode(course.id) for course in _get_all_courses()
-                          if has_access(user, access_type, course)]
+        if not course_ids:
+            if GlobalStaff().has_user(user):
+                # Global staff have access to all courses, so just return them all.
+                course_ids = [unicode(course.id) for course in _get_all_courses()]
+            else:
+                course_ids = list(UserBasedRole(user, access_type).courses_with_role()
+                                  .values_list('course_id', flat=True).distinct())
```

Comment:
Just double check with our production courses, and it only returns CourseDescriptors. @clintonb it is safe to change the line above to  :+1: 
```
    course_ids = [unicode(course.id) for course in modulestore().get_courses()]
```

@cpennington now I am not sure where the filtering came from. Probably I just followed what the branding app is doing. It seems like you wrote the line that does the filtering there, but almost two years ago.